src/node.c

```diff
@@ -95,10 +95,14 @@ void free_node(node_t *n) {
 	sockaddrfree(&n->address);
 
 #ifndef DISABLE_LEGACY
-	cipher_free(&n->incipher);
-	digest_free(&n->indigest);
-	cipher_free(&n->outcipher);
-	digest_free(&n->outdigest);
+	cipher_free(n->incipher);
+	n->incipher = NULL;
```

Comment:
Why is it necessary to NULL these variables if we are going to `free(n)` anyway?